src/main/java/org/sonarlint/intellij/ui/ruledescription/RuleHeaderPanel.kt

```diff
@@ -91,6 +98,47 @@ class RuleHeaderPanel : JBPanel<RuleHeaderPanel>(FlowLayout(FlowLayout.LEFT)) {
     fun update(ruleKey: String, type: RuleType, severity: IssueSeverity) {
         clear()
         updateCommonFields(type, ruleKey)
+        updateRuleSeverity(severity)
+    }
+
+    fun update(ruleKey: String, type: RuleType, severity: IssueSeverity, finding: Finding) {
+        clear()
+        updateCommonFields(type, ruleKey)
+        updateRuleSeverity(severity)
+        val serverFindingKey: String
+        val isIssueValid: Boolean
+
+        when (finding) {
+            is LiveIssue -> {
+                serverFindingKey= finding.serverFindingKey ?: return
+                isIssueValid = finding.isValid
+            }
+
+            is LocalTaintVulnerability -> {
+                serverFindingKey = finding.key() ?: return
+                isIssueValid = finding.isValid()
+            }
+
+            else -> {
+                return
+            }
+        }
+
+        openIssueDialog(serverFindingKey, isIssueValid)
```

Comment:
The name of the method might be confusing, we are not opening the dialog right away, we are setting the listener on the button